Sources/PointFree/Admin/EpisodeCredit.swift

```diff
@@ -0,0 +1,77 @@
+import Css
+import Either
+import Foundation
+import Html
+import HtmlCssSupport
+import HttpPipeline
+import HttpPipelineHtmlSupport
+import Optics
+import Prelude
+import Styleguide
+import Tuple
+
+let showEpisodeCreditsMiddleware: Middleware<StatusLineOpen, ResponseEnded, Tuple1<Database.User?>, Data> =
+  requireAdmin
+    <| writeStatus(.ok)
+    >-> respond(showEpisodeCreditsView.contramap(const(unit)))
+
+let addEpisodeCreditMiddleware: Middleware<StatusLineOpen, ResponseEnded, Tuple3<Database.User?, String?, Int?>, Data> =
+  requireAdmin
+    <<< filterMap(
+      over2(fetchUser(uuidString:)) >>> sequence2 >>> map(require2),
```

Comment:
intense, but it works. transforms a 

```swift
Tuple3<Database.User, String?, Int?>
```

into a 

```swift
Tuple3<Database.User, Database.User, Int?>
```

Reply:
Ah nice, makes sense! Wish we had row types to annotate.

Reply:
(One thought is to `map` to `UUID.init >>> User.Id.init` so it gets captured in the sig, but it's like a minor readability improvement.)

Reply:
done! b7696ed